libs/wire-api-federation/src/Wire/API/Federation/Error.hs

```diff
@@ -322,11 +323,12 @@ federationUnexpectedError msg =
     ("Could parse body, but got an unexpected error response: " <> LT.fromStrict msg)
 
 federationUnreachableError :: Set Domain -> Wai.Error
-federationUnreachableError (Set.map domainText -> ds) =
-  Wai.mkError
+federationUnreachableError (Set.toList -> ds) =
+  Wai.Error
     status
     "federation-unreachable-domains-error"
-    ("The following domains are unreachable: " <> (LT.pack . show . Set.toList) ds)
+    ("The following domains are unreachable: " <> (LT.pack . show . map domainText) ds)
+    (flip FederationErrorData T.empty <$> NE.nonEmpty ds)
```

Comment:
I haven't thought about it much, but I suppose it's OK to leave out the info on the path of the failing request in this case.